modules/gdscript/gdscript.cpp

```diff
@@ -2818,6 +2835,21 @@ String ResourceFormatLoaderGDScript::get_resource_type(const String &p_path) con
 	return "";
 }
 
+ResourceUID::ID ResourceFormatLoaderGDScript::get_resource_uid(const String &p_path) const {
+	String ext = p_path.get_extension().to_lower();
+
+	if (ext != "gd") {
+		return ResourceUID::INVALID_ID;
+	}
+
+	const GDScriptParser parser = GDScript::parse_file(GDScript::get_raw_source_code(p_path));
+	const GDScriptParser::ClassNode *c = parser.get_tree();
+	if (!c) {
+		return ResourceUID::INVALID_ID;
+	}
+	return ResourceUID::get_singleton()->text_to_id(c->uid_string);
+}
```

Comment:
It's probably worth caching the UID (don't parse the script if the UID is already assigned).

Reply:
I think the editor handles caching automatically.

Reply:
Then this is not necessary if the method is not exposed.